sycl/plugins/CMakeLists.txt

```diff
@@ -6,8 +6,11 @@ foreach(plugin ${SYCL_ENABLE_PLUGINS})
   add_subdirectory(${plugin})
 endforeach()
 
-# level_zero plugin depends today on unified_runtime plugin
+# level_zero, cuda, hip, and native_cpu plugins depends today on unified_runtime plugin
 # and unified_runtime plugin is not an independent plugin, adding it explicitly
-if ("level_zero" IN_LIST SYCL_ENABLE_PLUGINS)
+if ("level_zero" IN_LIST SYCL_ENABLE_PLUGINS OR
+        "hip" IN_LIST SYCL_ENABLE_PLUGINS OR
+        "cuda" IN_LIST SYCL_ENABLE_PLUGINS OR
+        "native_cpu" IN_LIST SYCL_ENABLE_PLUGINS)
   add_subdirectory(unified_runtime)
```

Comment:
Further testing uncovered that `add_subdirectory(...)` does not work here correctly when called twice.

- Should we handle the situation that `SYCL_ENABLE_PLUGINS` already contains `unified_runtime`?
- If yes, would it be better to 
  - extend an `if` here, or 
  - add `include_guard(GLOBAL)` to `unified_runtime/CMakeLists.txt`?

I lean towards "Yes" and "include_global".

Reply:
Turns out, include_guard does not work the way I thought it works. Added an extra layer of `if` to avoid including the directory twice in case user specified something like `-DSYCL_ENABLE_PLUGINS='hip;unified_runtime'`